doc/semtech_radio_refactor.md

```diff
@@ -0,0 +1,131 @@
+# Semtech Radio Refactor
+
+The Sidewalk Semtech radio refactor is based on "LBM_Zephy" LoRa Basics Modem integration in Zephyr OS.
+Project URL: https://github.com/Lora-net/LBM_Zephyr
+
+## Semtech DTS and Sidewalk Driver Structure Mapping
+
+This document describes the complete mapping between Semtech radio DTS (Device Tree Source) configurations and the Sidewalk driver structures for SX126X radio modules. The implementation properly handles both required and optional DTS properties with appropriate default values.
+
+### DTS Properties Analysis
+
+#### Required Properties (✅ Mapped)
+| DTS Property | Driver Structure Field | Status | Description |
+|--------------|----------------------|--------|-------------|
+| `reset-gpios` | `gpio_power` | ✅ Mapped | GPIO for radio reset signal |
+| `busy-gpios` | `gpio_radio_busy` | ✅ Mapped | GPIO for radio busy signal |
+| `reg-mode` | `regulator_mode` | ✅ Mapped | Power regulator mode (LDO/DCDC) |
+| `tcxo-voltage` | `tcxo.voltage` | ✅ Mapped | TCXO supply voltage |
+| `tcxo-wakeup-time` | `tcxo.timeout` | ✅ Mapped | TCXO wakeup time in ms |
+
+#### Optional Properties (✅ Mapped with Defaults)
+| DTS Property | Driver Structure Field | Default | Status | Description |
+|--------------|----------------------|---------|--------|-------------|
+| `dio1-gpios` | `gpio_int1` | `INVALID_DT_GPIO` | ✅ Mapped | GPIO for DIO1 interrupt signal |
+| `spi-max-frequency` | `bus_selector.speed_hz` | `8MHz` | ✅ Mapped | SPI bus frequency |
+| `cs-gpios` (from arduino_spi) | `bus_selector.client_selector` | `INVALID_DT_GPIO` | ✅ Mapped | SPI chip select GPIO |
+| `rx-boosted` | `rx_boost` | `false` | ✅ Mapped | RX boosted mode configuration |
+
+#### Optional Properties (❌ Not Implemented in Driver Structure)
+| DTS Property | Reason | Description |
+|--------------|--------|-------------|
+| `dio2-gpios` | No field in driver structure | DIO2 as generic IRQ line |
+| `dio2-as-rf-switch` | No field in driver structure | DIO2 as RF switch control |
+| `dio3-gpios` | No field in driver structure | DIO3 as generic IRQ line |
+| `dio3-as-tcxo-control` | No field in driver structure | DIO3 as TCXO control |
+| `xtal-capacitor-value-xta` | No field in driver structure | XTA trimming capacitor value |
+| `xtal-capacitor-value-xtb` | No field in driver structure | XTB trimming capacitor value |
+| `tx-power-offset` | No field in driver structure | TX power offset configuration |
```

Comment:
OK this value is really not used in sidewalk driver, so it is correct to say that it will be ignored.